components/motor/gpio/basic.go

```diff
@@ -186,39 +183,51 @@ func (m *Motor) setPWM(ctx context.Context, powerPct float64, extra map[string]i
 		errs = multierr.Combine(errs, m.EnablePinHigh.Set(ctx, true, extra))
 	}
 
+	m.powerPct = powerPct
+	powerPct = math.Min(powerPct, m.maxPowerPct)
+	powerPct = math.Max(powerPct, -1*m.maxPowerPct)
+
```

Comment:
I think this does what we want. Checks the max & min powers, then updates the motor's m.powerPct

```suggestion
	powerPct = sign(powerPct)*math.Min(math.Abs(powerPct), m.maxPowerPct) // check max power
	powerPct = sign(powerPct)*math.Max(math.Abs(powerPct), m.minPowerPct) // check min power
	m.powerPct = powerPct
```